utils/wxrc/wxrc.cpp

```diff
@@ -626,41 +627,105 @@ void XmlResApp::MakePackageZIP(const wxArrayString& flist)
 
 static wxString FileToCppArray(wxString filename, int num)
 {
+    // [2024-03-15]
+    //   Some of the `wxString` objects are defined at the beginning of the function
+    //   (before the actual use, and with larger scope than needed)
+    //   in order to avoid repeated allocations and deallocations.
     wxString output;
     wxString tmp;
     wxString snum;
     wxFFile file(filename, wxT("rb"));
     wxFileOffset offset = file.Length();
     wxASSERT_MSG( offset >= 0 , wxT("Invalid file length") );
 
-    const size_t lng = wx_truncate_cast(size_t, offset);
+    // [2024-03-15]
+    //
+    //   There are two environments (processes) to consider:
+    //     - the resource compiler;
+    //     - the process using the resource.
+    //
+    //   Previously, we used to check whether `offset` fitted into `size_t`.
+    //
+    //   But this used to check against the `size_t` of the resource compiler.
+    //   The check might have passed, but this might have been misleading
+    //   (if the resource compiler is a 64-bit process and the process using the resource is a 32-bit process).
+    //
+    //   Also, for Windows at least, the file format for modules (executables and shared libraries)
+    //   which are to be loaded in 64-bit processes
+    //   is not called PE64, but instead PE32+,
+    //   and yes, in this format virtual addresses are indeed 64-bit,
+    //   but in fact relative virtual addresses (and also file offsets) are 32-bit
+    //   (i.e. a single module cannot span for more than 4 GiB in virtual memory).
+    //
+    //   It follows that the check might have also been misleading
+    //   in the case of the resource compiler being a 64-bit Linux process
+    //   and the process using the resource being a Windows process -- even a 64-bit Windows process.
+    //
+    //   Therefore, we now check whether the size of the resource fits
+    //   not into `size_t`, but instead into `uint32_t`,
+    //   i.e. we do not support resources larger than 4 GiB.
+    //
+    const wxUint32 lng = wx_truncate_cast(wxUint32, offset);
```

Comment:
Thanks, I agree with the fix, but it's unrelated to the other changes and should be in a separate commit.

Reply:
Thank you.

We now have https://github.com/wxWidgets/wxWidgets/pull/24437 as partial replacement for this pull request (not addressing the main topic) -- for review and suggestions for modifications.

I am going to get back a little later these days with that and also with the rest (the main topic of this pull request is not addressed yet by https://github.com/wxWidgets/wxWidgets/pull/24437).

Reply:
> Thanks, I agree with the fix, but it's unrelated to the other changes and should be in a separate commit.

https://github.com/wxWidgets/wxWidgets/pull/24437/commits/6952680b6e7dfbef5957372df75c118349d811be

Reply:
#24437 (with smaller commits) now fully replaces this pull request...